src/components/UserProfile/UserSideBar.jsx

```diff
@@ -25,81 +81,59 @@ const USER_INFO_DOM_ELEMENTS = [
   },
 ];
 
-const UserSideBar = ({ user }) => {
-  let userInfoDOM = USER_INFO_DOM_ELEMENTS.map(elem => {
-    if (user[elem.schemaKey] && user[elem.schemaKey].length > 0) {
+
+const UserInfo = ({ user }) => {
+  return USER_INFO_DOM_ELEMENTS.map((elem, idx) => {
+    const userComponent = ({ data }) => {
       return (
         <div className={elem.divClassName}>
           <h1 className="user-sidebar-subcategory">{elem.desc}</h1>
-          <p>{user[elem.schemaKey]}</p>
+          <p>{data}</p>
         </div>
       )
     }
-    else {
-      return null;
-    }
-  });
-
-  let skillDOM = null;
-  if (user.skills && user.skills.length > 0) {
-    skillDOM = (
-      <div className="user-skills">
-        <h1 className="user-sidebar-subcategory">skills</h1>
-        <ul>
-          {user.skills.map(elem => (<li>{elem}</li>))}
-        </ul>
-      </div>
-    )
-  }
-
-  // once links are integrated again, render this
-  // check for fb/linkedin/gitub
-  let linkDOM = null;
-  if (user) {
-    linkDOM = (
-      <div className="user-links">
-        <h1 className="user-sidebar-subcategory">links</h1>
-        <ul>
-          <li>
-            <a target="_blank" href={"https://www.github.com/" + user.username}>
-              <i className="fab fa-github fa-3x" />
-            </a>
-          </li>
-        </ul>
-      </div>
+    return (
+      <EditableTextField
+        key={idx}
+        large
+        mutation={ userUpdate }
+        mutationName="userUpdate"
+        mutationInputName="user_data"
+        fieldName={elem.schemaKey}
+        fieldData={user[elem.schemaKey]}
+        hasPermission={true}
```

Comment:
needs to check if viewing user is the same as the users profile (has edit rights)
write a boolean returning function to be called in hasPermission